lib/src/components/layout/accordion.dart

```diff
@@ -20,39 +20,149 @@ class _AccordionState extends State<Accordion> {
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
     final scaling = theme.scaling;
+    final accTheme = ComponentTheme.maybeOf<AccordionTheme>(context) ??
+        const AccordionTheme();
     return Data.inherit(
         data: this,
         child: IntrinsicWidth(
-          child: Column(
-              mainAxisSize: MainAxisSize.min,
-              crossAxisAlignment: CrossAxisAlignment.stretch,
-              children: [
-                ...join(
-                    widget.items,
-                    ShadcnAnimatedContainer(
-                      duration: kDefaultDuration,
-                      color: theme.colorScheme.muted,
-                      height: 1 * scaling,
-                    )),
-                const Divider(),
-              ]),
+          child: ComponentTheme(
+            data: accTheme,
+            child: Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  ...join(widget.items.map((item) {
+                    return item is AccordionItem
+                        ? item._applyTheme(accTheme)
+                        : item;
+                  }),
+                      ShadcnAnimatedContainer(
+                        duration: kDefaultDuration,
+                        color: theme.colorScheme.muted,
+                        height: accTheme.dividerHeight * scaling,
+                      )),
+                  const Divider(),
+                ]),
+          ),
         ));
   }
 }
 
+/// {@template accordion_theme}
+/// Styling options for [AccordionItem], [AccordionTrigger] and [Accordion].
+/// {@endtemplate}
+class AccordionTheme {
+  /// Duration of the collapse/expand animation.
+  final Duration duration;
+
+  /// Curve of the animation (played when expanding).
+  final Curve curve;
+
+  /// Reverse curve of the animation (played when collapsing).
+  final Curve reverseCurve;
+
+  /// The gap between the trigger and the content (or other triggers if collapsed).
+  ///
+  /// The padding is applied to the top and bottom of the trigger.
+  final double padding;
+
+  /// The gap between the trigger text and the icon.
+  final double iconGap;
+
+  /// The height of the divider between each item.
+  final double dividerHeight;
+
+  /// {@macro accordion_theme}
+  const AccordionTheme({
+    this.duration = const Duration(milliseconds: 200),
+    this.curve = Curves.easeIn,
+    this.reverseCurve = Curves.easeOut,
+    this.padding = 16,
+    this.iconGap = 18,
+    this.dividerHeight = 1,
```

Comment:
Please refer back to the [guidelines](https://github.com/sunarya-thito/shadcn_flutter/issues/115), Component theme data class should not have default values